src/main/resources/category/java/remoting.xml

```diff
@@ -408,535 +408,584 @@ return HttpClientBuilder.create()
+        return factory;
+    }
 }
 ]]>
-</example>
-</rule>
+        </example>
+    </rule>
 
-<rule -name="AvoidHttpHostOneArgumentConstructor"
-class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
-language="java"
-message="Avoid the HttpHost constructor with only hostname, provide port number and protocol as well."
-externalInfoUrl="${doc_root}/JavaCodePerformance.md#ibi16">
-<description>Problem: the HttpHost constructor with one argument must only be provided with a host name, the default port 80 and protocol http are implied.
-The mistake of providing a URL and assuming it will be parsed into hostname, port and protocol is easily made.
-When this HttpHost is then used for a route and stored socketConfig for, port 80 is added for the host and the socketConfig is stored with the wrong key and will not be used.
-It is typically difficult to find out if the config is actually used. Note that http://github.com/jborgers/http-client-monitor helps here.&#13;
-Solution: Use the HttpHost constructor with 2 (including port) or preferably 3 arguments (including port and protocol). (jpinpoint-rules)</description>
-<priority>2</priority>
-<properties>
-<property name="tags" value="jpinpoint-rule,confusing,suspicious" type="String" description="classification"/>
-<property name="xpath">
-    <value><![CDATA[
-//AllocationExpression[pmd-java:typeIs("org.apache.http.HttpHost")]/Arguments/ArgumentList[count(./Expression) = 1]/../..
+
+    <rule name="AvoidHttpHostOneArgumentConstructor"
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          language="java"
+          message="Avoid the HttpHost constructor with only hostname, provide port number and protocol as well."
+          externalInfoUrl="${doc_root}/JavaCodePerformance.md#ibi16">
+        <description>Problem: the HttpHost constructor with one argument must only be provided with a host name, the
+            default port 80 and protocol http are implied.
+            The mistake of providing a URL and assuming it will be parsed into hostname, port and protocol is easily
+            made.
+            When this HttpHost is then used for a route and stored socketConfig for, port 80 is added for the host and
+            the socketConfig is stored with the wrong key and will not be used.
+            It is typically difficult to find out if the config is actually used. Note that
+            http://github.com/jborgers/http-client-monitor helps here.&#13;
+            Solution: Use the HttpHost constructor with 2 (including port) or preferably 3 arguments (including port and
+            protocol). (jpinpoint-rules)
+        </description>
+        <priority>2</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,confusing,suspicious" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//ConstructorCall[pmd-java:typeIs("org.apache.http.HttpHost")]/ArgumentList[@Size=1]/..
```

Comment:
can also place ArgumentList check inside brackets `[...]` to avoid `/..` at end?

Reply:
yes, shorter, adjusted